packages/react/src/components/MultiSelect/FilterableMultiSelect.tsx

```diff
@@ -0,0 +1,1018 @@
+  const [inputFocused, setInputFocused] = useState<boolean>(false);
+
+  const {
+    selectedItems: controlledSelectedItems,
+    onItemChange,
+    clearSelection,
+  } = useSelection({
+    disabled,
+    initialSelectedItems,
+    onChange,
+    selectedItems: selected,
+  });
+
+  const textInput = useRef<HTMLInputElement>(null);
+  const filterableMultiSelectInstanceId = useId();
+
+  const prefix = usePrefix();
+
+  if (prevOpen !== open) {
+    setIsOpen(open);
+    setPrevOpen(open);
+  }
+
+  const sortedItems = sortItems(
+    filterItems(items, { itemToString, inputValue }),
+    {
+      selectedItems: {
+        top: controlledSelectedItems,
+        fixed: [],
+        'top-after-reopen': topItems,
+      }[selectionFeedback],
+      itemToString,
+      compareItems,
+      locale,
+    }
+  );
+
+  const inline = type === 'inline';
+  const showWarning = !invalid && warn;
+
+  const wrapperClasses = cx(
+    `${prefix}--multi-select__wrapper`,
+    `${prefix}--multi-select--filterable__wrapper`,
+    `${prefix}--list-box__wrapper`,
+    containerClassName,
+    {
+      [`${prefix}--multi-select__wrapper--inline`]: inline,
+      [`${prefix}--list-box__wrapper--inline`]: inline,
+      [`${prefix}--multi-select__wrapper--inline--invalid`]: inline && invalid,
+      [`${prefix}--list-box__wrapper--inline--invalid`]: inline && invalid,
+      [`${prefix}--list-box--up`]: direction === 'top',
+      [`${prefix}--list-box__wrapper--fluid--invalid`]: isFluid && invalid,
+      [`${prefix}--list-box__wrapper--fluid--focus`]: isFluid && isFocused,
+      [`${prefix}--list-box__wrapper--slug`]: slug,
+    }
+  );
+  const helperId = !helperText
+    ? undefined
+    : `filterablemultiselect-helper-text-${filterableMultiSelectInstanceId}`;
+  const labelId = `${id}-label`;
+  const titleClasses = cx({
+    [`${prefix}--label`]: true,
+    [`${prefix}--label--disabled`]: disabled,
+    [`${prefix}--visually-hidden`]: hideLabel,
+  });
+  const helperClasses = cx({
+    [`${prefix}--form__helper-text`]: true,
+    [`${prefix}--form__helper-text--disabled`]: disabled,
+  });
+  const inputClasses = cx({
+    [`${prefix}--text-input`]: true,
+    [`${prefix}--text-input--empty`]: !inputValue,
+    [`${prefix}--text-input--light`]: light,
+  });
+  const helper = helperText ? (
+    <div id={helperId} className={helperClasses}>
+      {helperText}
+    </div>
+  ) : null;
+  const menuId = `${id}__menu`;
+  const inputId = `${id}-input`;
+
+  useEffect(() => {
+    if (!isOpen) {
+      setTopItems(controlledSelectedItems);
+    }
+  }, [controlledSelectedItems, isOpen, setTopItems]);
+
+  function handleMenuChange(forceIsOpen: boolean): void {
+    const nextIsOpen = forceIsOpen ?? !isOpen;
+    setIsOpen(nextIsOpen);
+    if (onMenuChange) {
+      onMenuChange(nextIsOpen);
+    }
+  }
+
+  // function handleOnOuterClick() {
+  //   handleOnMenuChange(false);
+  // }
```

Comment:
@gui just want to double check this is no longer needed? Maybe we should add support for onChange to be logged to the actions pane in one of the storybook stories so this is testable in the deploy preview?